### ISSUE TITLE
Add specific page titles for user profile pages and other views that lack them

### ISSUE THREAD

**Issue by ragesoss** [WikiEducationFoundation/WikiEduDashboard#4616]
### Current Behavior:
Many pages on the Dashboard — such as a user profile page like https://dashboard.wikiedu.org/users/Sage%20(Wiki%20Ed) — don't have specific page titles (ie, what shows up in browser history and on the browser tab). This makes it harder to navigate through the site using browser history.

### Desired Behavior:
User profile pages and many other pages should follow the same convention already in place for course pages, with a `before_title` string added to the standard title. For example, a user profile page should have a title like: `User:Sage_(Wiki_Ed) — Wiki Education Dashboard` instead of just `Wiki Education Dashboard`.

### Related Files & Code:

For an example of where this is already implemented, for course pages, see `app/views/courses/shows.html.haml`.

**Comment by ragesoss** [comment]
Some of these, including user profile pages, have been fixed, but there are plenty of others that still don't have useful titles.

The training pages — like https://dashboard.wikiedu.org/training and https://dashboard.wikiedu.org/training/students/wikipedia-essentials and https://dashboard.wikiedu.org/training/students/wikipedia-essentials/policies-and-guidelines-basic-overview — are one area that needs good titles.

**Comment by ragesoss** [comment]
@chepyego hi! I just recommended this for another newcomer to the project. If you'd like to find something else to work on, I can look around for something. Let me know what kind of issue you're interested in.

**Comment by TheTrio** [comment]
For some reason, my training contents page is entirely empty. 

![image](https://user-images.githubusercontent.com/10794178/153591254-36aa7d4a-db22-4ecf-a9fa-86b93acd567e.png)

I did find a `training-content` folder in the root of the project but I have no idea how make that show up on the website. Was this some configuration I missed?

**Comment by ragesoss** [comment]
That takes care of the server-rendered part of the training pages. A possible further improvement would be to add title handling into the React app so that titles change as you navigate through the different tabs of a course page, and through the individual slides of a training module.

Some other pages that still lack titles:
* /admin
* /faq and (eg) /faq/1

**Comment by TheTrio** [comment]
Yes, that should be doable.

**Comment by TheTrio** [comment]
> through the individual slides of a training module.

Since `slideTitle` is already a state we managed, it should be straightforward to just prepend that to the page title.  

>  so that titles change as you navigate through the different tabs of a course page

I'm wondering what's the best way to do this. Since there is no state here (as far as I can tell, everything is being managed by React Router), I can't find a clean way to do this. We do have access to the current location though - so I could write a switch statement which sets the title depending on what tab we're on. But that doesn't seem like a very good idea - though I'm unable to think of anything better. 

Thoughts?

**Comment by saha23s** [comment]
Hi @ragesoss , I am just curious if there are more pages lacking titles to work on for this issue.

**Comment by ragesoss** [comment]
@saha23s yes, there are a number of them. A few I found with a quick scan include all the FAQ routes, as well as the /admin index page and some of the pages linked from there.

**Comment by saha23s** [comment]
@ragesoss I am aiming to make the title look like this: 
For admin pages: Username - Programs and Events Dashboard
For FAQ pages: Username - Title of FAQ - Programs and Events Dashboard
Will it work?

**Comment by ragesoss** [comment]
@saha23s username is not relevant in most cases, including those ones.

I'd say something like 'Admin tools' is a good name for the admin, and perhaps 'FAQ: [title of FAQ] - [Dashboard title]' for the FAQ pages.

**Comment by saha23s** [comment]
@ragesoss I have created this pull request for only admin pages. [ I think after working with the FAQ issue, I can update the titles of faq pages ]
https://github.com/WikiEducationFoundation/WikiEduDashboard/pull/5322 
This is my first PR editing codebase, I don't understand why there's 5 commits showing in the PR instead of 1 commit? How can I fix this?
![image](https://user-images.githubusercontent.com/93749279/227225244-2e40a83b-553d-4a46-8880-445443fa8f7f.png)

**Comment by saha23s** [comment]
@ragesoss I have created a PR for the admin pages. Also, a suggestion is that for a couple of pages, I saw that the keyword for that page is after the default title - which basically does not help the user to distinguish between pages in the open browser tabs. I think it's better to add the keyword before the title. Just a note if anyone works on this issue as well!

**Comment by HARSHITHASALADI1** [comment]
Could you please assign me this task?

**Comment by fillingtothemomo** [comment]
@ragesoss is this issue still active?

**Comment by ragesoss** [comment]
@fillingtothemomo I think so. I don't know which pages still lack titles, but I'm pretty sure there are a few.

**Comment by Solobarine** [comment]
I would like to work on this @ragesoss

**Comment by Solobarine** [comment]
@ragesoss I have added descriptive titles for some pages. Here is the pull request with the changes: [[PR](https://github.com/WikiEducationFoundation/WikiEduDashboard/pull/5977)](https://github.com/WikiEducationFoundation/WikiEduDashboard/pull/5977)

**Comment by Jay2006sawant** [comment]
@ragesoss I am new to Open Source Contribution,I would like to contribute on this issue.Would you please assign it to me?

**Comment by ragesoss** [comment]
@Jay2006sawant we don't assign issues, but feel free to work on this. Please only open one PR at at time, though.

**Comment by Jay2006sawant** [comment]
@ragesoss  is this issue still active? Also any idea which pages still lacks title?

**Comment by pablo1973ar** [comment]
I'll work on this issue

**Comment by BrenSerena** [comment]
I'll work on this issue (alerts list, alert followup)
